### PR TITLE
[CI/CD][workers][SIMS #1442]Feature/#1442 cra bypass

### PR DESCRIPTION
**Pull request by andrewsignori-aot**
- Added the CRA bypass to the workers. The bypass was removed from the API.
- Camunda upgraded to 8.1 in the Cloud and local dev env (docker-compose). The workflow was also upgraded to version 8.1. 

**Actions items needed**: Please ensure to update the desktop modeler to the latest version.

### TODO
- CRA bypass 👍
- Zebee maxJobsToActivate configuration for all workers 👍 
- Remove hardcoded timeout for messages (`timeToLive: Duration.seconds.of(30)`) 👍 - created a constant in workers app.
- Test Camunda inclusive gateway on Csamunda 8.1 👎 - inclusive gateways were not used because it is partially implemented only on version 8.1
- Reassessment tests 👍 

Upcoming PR to remove non-used methods from API.

### CODE REVIEW

File: configs/env-example [hunk 1]
```diff
@@ -1,3 +1,6 @@
+# Set time zone for the same expected from Openshift.
+TZ=UTC
```
Comment (ann-aot) [inline configs/env-example:2]: 👍

File: sources/packages/backend/libs/services/src/constants/system-configurations-constants.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import { Duration } from "zeebe-node";
+
+/**
+ * Default time that a message can wait, once it is delivery to the broker,
+ * to find its correlated key.
+ */
+export const ZEEBE_PUBLISH_MESSAGE_DEFAULT_TIME_TO_LEAVE =
```
Comment (ann-aot) [inline sources/packages/backend/libs/services/src/constants/system-configurations-constants.ts:7]: mm...what is this for. I mean do we need a default time? and what happens if it exceeds the default time

File: sources/packages/backend/apps/workers/src/types/max-jobs-to-activate.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
+/**
+ * Suggested values to be considered while defining the
+ * amount of jobs that a worker can request to be activated.
+ * !The best choice for every scenario can be defined only after
+ * !some benchmarks are acquired. The below options and descriptions
+ * !are just a suggestion to have a start point. As a general recommendation
+ * !activating one job per call does not seems a recommended practice.
+ */
+export enum MaxJobsToActivate {
```
Comment (ann-aot) [inline sources/packages/backend/apps/workers/src/types/max-jobs-to-activate.ts:9]: mm... how do we decide the count for a worker?

File: sources/camunda-docker-compose-core.yml [hunk 4]
```diff
@@ -91,9 +102,24 @@ services:
     networks:
       - camunda-platform
 
+  kibana:
+    image: docker.elastic.co/kibana/kibana:${ELASTIC_VERSION:-7.17.0}
```
Comment (ann-aot) [inline sources/camunda-docker-compose-core.yml:106]: `ELASTIC_VERSION` here and 78 are different

File: sources/camunda-docker-compose-core.yml [hunk 5]
```diff
@@ -63,8 +63,19 @@ services:
...
+      - ZEEBE_ADDRESS=zeebe:26500
+      - ZEEBE_INSECURE=true
+    networks:
+      - camunda-platform
+    depends_on:
+      - zeebe
+
   elasticsearch: # https://hub.docker.com/_/elasticsearch
-    image: docker.elastic.co/elasticsearch/elasticsearch:${ELASTIC_VERSION:-7.17.0}
+    image: docker.elastic.co/elasticsearch/elasticsearch:${ELASTIC_VERSION:-7.17.5}
```
Comment (ann-aot) [inline sources/camunda-docker-compose-core.yml:78]: why did camunda need elastic search again

File: sources/packages/backend/apps/workers/src/services/cra-income-verification/cra-income-verification.service.ts [hunk 6]
```diff
@@ -46,6 +52,40 @@ export class CRAIncomeVerificationService extends RecordDataModelService<CRAInco
...
 
+  /**
+   * This method allows the simulation of a complete cycle of the CRA send/response
+   * process that allows the workflow to proceed without the need of the actual
+   * CRA verification happens. This is enabled based in a environment variable,
+   * that is by default disabled and should be enabled only for DEV purposes on
+   * local developer machine or on an environment where the CRA process is not enabled.
+   * !This code should not be executed on production.
+   * @param verificationId CRA verification id waiting to be processed.
+   * @returns update result.
```
Comment (ann-aot) [inline sources/packages/backend/apps/workers/src/services/cra-income-verification/cra-income-verification.service.ts:63]: promise is not retuning

File: sources/camunda-docker-compose-core.yml (hunk 5 above)
Comment (dheepak-aot) [reply sources/camunda-docker-compose-core.yml:78]: As per the previous PR conversation, I believe elastic search is required for the prometheus event monitoring. 
@andrewsignori-aot correct me if that is wrong.

File: configs/env-example [hunk 7]
```diff
@@ -1,3 +1,6 @@
...
+
 # Project Specific Env
```
Comment (dheepak-aot) [inline configs/env-example:4]: Can we add BYPASS_CRA_INCOME_VERIFICATION=false in the sample env file(may be with a comment) which will be helpful for any new person setting up the application locally.

File: sources/camunda-docker-compose-core.yml (hunk 4 above)
Comment (andrewsignori-aot) [reply sources/camunda-docker-compose-core.yml:106]: This docker-compose comes from the below source. We are not changing it just consuming it unless there is something really needed to be changed. 
https://github.com/camunda/camunda-platform/tree/8.1.2

File: sources/camunda-docker-compose-core.yml (hunk 5 above)
Comment (andrewsignori-aot) [reply sources/camunda-docker-compose-core.yml:78]: This docker-compose comes from the below source. We are not changing it just consuming it unless there is something really needed to be changed. 
https://github.com/camunda/camunda-platform/tree/8.1.2

File: sources/packages/backend/apps/workers/src/services/cra-income-verification/cra-income-verification.service.ts (hunk 6 above)
Comment (andrewsignori-aot) [reply sources/packages/backend/apps/workers/src/services/cra-income-verification/cra-income-verification.service.ts:63]: It does not need to, right?

File: configs/env-example (hunk 7 above)
Comment (andrewsignori-aot) [reply configs/env-example:4]: Comments added.

File: sources/packages/backend/apps/workers/src/types/max-jobs-to-activate.ts (hunk 3 above)
Comment (andrewsignori-aot) [reply sources/packages/backend/apps/workers/src/types/max-jobs-to-activate.ts:9]: Nothing too concrete right now. Without executing some benchmarks there is no way to be precise but we can try to use the examples in place right now.
For instance, take a look at `MSFAA` vs `Update Application Status` and what was defined to them.
It is just a matter to have a "best guess" right now.
The most important, IMO, would be not left as just 1 job to be activated per time. Open to suggestions.

File: sources/camunda-docker-compose-core.yml (hunk 5 above)
Comment (ann-aot) [reply sources/camunda-docker-compose-core.yml:78]: mm...okay .. so for monitoring and visualization Prometheus (like grafana) is used, which uses elastic search, and for elastic search monitoring it uses kibana..

File: sources/packages/backend/apps/workers/src/services/cra-income-verification/cra-income-verification.service.ts (hunk 6 above)
Comment (ann-aot) [reply sources/packages/backend/apps/workers/src/services/cra-income-verification/cra-income-verification.service.ts:63]: ya...so the comment is also not needed right

File: sources/camunda-docker-compose-core.yml (hunk 5 above)
Comment (andrewsignori-aot) [reply sources/camunda-docker-compose-core.yml:78]: Right now the only applications that I really concerned about and using are Zeebe and Operate while running on my local machine. While in the cloud we also have "Optimize" for performance improvements, but it was not explored yet.

image

File: sources/packages/backend/apps/workers/src/services/cra-income-verification/cra-income-verification.service.ts (hunk 6 above)
Comment (andrewsignori-aot) [reply sources/packages/backend/apps/workers/src/services/cra-income-verification/cra-income-verification.service.ts]: Now I got it 😄

Comment (andrewsignori-aot) [reply sources/packages/backend/apps/workers/src/services/cra-income-verification/cra-income-verification.service.ts]: Comment removed.

File: sources/packages/backend/apps/workers/src/types/max-jobs-to-activate.ts [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+  /**
+   * Used for operations where the regular execution, considering
+   * all the activated jobs, could take a second or a couple of seconds.
+   */
+  Normal = 10,
+  /**
+   * Used for simple operations where the regular execution, considering
+   * all the activated jobs, could take few milliseconds to a second or slightly more.
+   */
+  Hight = 15,
```
Comment (guru-aot) [inline sources/packages/backend/apps/workers/src/types/max-jobs-to-activate.ts:35]: is this high? Typo?

**Review by dheepak-aot — APPROVED** [review]
LGTM @andrewsignori-aot 👍

**Review by ann-aot — APPROVED** [review]
Good job @andrewsignori-aot 👍

**Review by guru-aot — APPROVED** [review]
LGTM, nice work on the upgrade @andrewsignori-aot

**Pull request merged**